### PR TITLE
openblas: 0.3.21 -> 0.3.24

### PR DESCRIPTION
**Pull request by markuskowa**
## Description of changes

Versions 0.3.22 and 0.3.23 had build problems.
Changelog: https://github.com/xianyi/OpenBLAS/releases

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by proofconstruction** [comment]
Build was failing on aarch64-darwin due to broken curl, but this was fixed by https://github.com/NixOS/nixpkgs/pull/255630.

With this change on the latest master, openblas builds fine on x86_64-linux, aarch64-linux, and aarch64-darwin.

This just needs a rebase and it's good to go.

**Comment by markuskowa** [comment]
@GrahamcOfBorg build openmolcas cp2k

**Comment by proofconstruction** [comment]
The rebase builds successfully on aarch64-darwin and aarch64-linux.

**Comment by markuskowa** [comment]
Are we good to merge?

**Review by wegank — APPROVED** [review]

File: pkgs/development/libraries/science/math/openblas/default.nix [hunk 1]
```diff
@@ -141,15 +141,15 @@ let
...
 
   src = fetchFromGitHub {
     owner = "xianyi";
     repo = "OpenBLAS";
     rev = "v${version}";
-    sha256 = "sha256-F6cXPqQai4kA5zrsa8E0Q7dD9zZHlwZ+B16EOGNXoXs=";
+    sha256 = "sha256-IuXhrZRB3o7kbnivv/6En/aAeF2F18sQw9pKs1WEJc4=";
   };
 
   patches = lib.optionals stdenv.hostPlatform.isLoongArch64 [
```
Comment (wegank) [inline pkgs/development/libraries/science/math/openblas/default.nix:155]: Can this patch be applied unconditionally?

Comment (markuskowa) [reply pkgs/development/libraries/science/math/openblas/default.nix:155]: I am not sure. The question is, if it still needed at all (see https://github.com/OpenMathLib/OpenBLAS/pull/3626#issuecomment-1735146632).
@wegank can you test it on LongArch64?

Comment (wegank) [reply pkgs/development/libraries/science/math/openblas/default.nix:155]: The loongarch64 cross [seems broken](https://hydra.nixos.org/eval/1799599?filter=loongarch64&compare=1799523&full=), and part of this patch doesn't apply. I think I'll leave it as is until Gentoo upgrades to 0.3.24.

**Pull request merged**